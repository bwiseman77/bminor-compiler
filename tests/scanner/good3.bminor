x+2
a: array [5] integer = {1, 2, 3, 4};
a: auto = 10;
/* more stuff */
print "The temperature is: ", temp, " degrees\n";
