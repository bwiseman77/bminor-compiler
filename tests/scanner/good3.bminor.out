-------------------------------------------
           token type  text
-------------------------------------------
               IDENT  x
     INTEGER_LITERAL  +2
               IDENT  a
               COLON  :
               ARRAY  array
 LEFT_SQUARE_BRACKET  [
     INTEGER_LITERAL  5
RIGHT_SQUARE_BRACKET  ]
             INTEGER  integer
              ASSIGN  =
        LEFT_BRACKET  {
     INTEGER_LITERAL  1
               COMMA  ,
     INTEGER_LITERAL  2
               COMMA  ,
     INTEGER_LITERAL  3
               COMMA  ,
     INTEGER_LITERAL  4
       RIGHT_BRACKET  }
           SEMICOLON  ;
               IDENT  a
               COLON  :
                AUTO  auto
              ASSIGN  =
     INTEGER_LITERAL  10
           SEMICOLON  ;
  MULTI_LINE_COMMENT  
               PRINT  print
      STRING_LITERAL  The temperature is: 
               COMMA  ,
               IDENT  temp
               COMMA  ,
      STRING_LITERAL   degrees

           SEMICOLON  ;
